docs/potato_paper.md

```diff
@@ -0,0 +1,111 @@
+# Potato Paper
+
+## Proof-of-Potato
+
+Proof-of-Potato (PoP) is a protocol for running decentralized sharded off-chain state machines.
+PoP is managed by a potato smart contract hosted on a secure blockchain such as Ethereum.
+The potato smart contract manages potato shard ownership and list of potato verifiers.
+Each potato shard runs a blockchain based potato state machine using our novel Proof-of-Tater (PoT) potato consensus algorithm.
+Other potato consensus protocols are possible too.
+We choose PoT because it offers lightweight security, speed, simplicity, and energy-efficient consensus...
+Just like a potato.
+
+### Potato Contract
+
+The potato contract is a potato smart contract.
+ The blockchain hosting the potato contract is called the potato main chain.
+
+
+**Potato Verifiers**:
+ Users aspiring to be potato verifiers can stake in potatoes in the PC to become farmers.
+ The contract manages a sorted list of all stakes over all time and has a public verifiable function assigning each potato shard to 5 potato verifiers.
+
+
+**Potato Shards**:
+ To prevent sybil attacks and manage economies in cross-potato shard transaction enabled systems,
+ it’s necessary to have a registry of all potato shards on the potato main chain.
+ The potato shard registry can also be used to store and modify additional potato data such as who is the owner, hash of a recent plant state, list of users, etc.
+
+
+### Potato Shards
+
+Potato Shards are operated by potato verifiers assigned by the potato contract.
+ Anyone can query the potato main chain for this info.
+ Shards have the benefits of permissioned blockchains, where the potato verifiers are determined by permissionless potato contract henc making each shard permissionless.
+ With this protocol, there is a lot of flexibility in state machine and consensus algorithm choice.
+
+
+**Proof-Of-Tater**:
+ For an applications such as games, there may be many potato shards either inactive or holding little value so we expect for verifiers to be offline.
+ This makes a consensus algorithm like PBFT inappropriate.
+ Instead we use our own novel permissioned Proof-of-Tater consensus algorithms which tolerates offline potato verifiers.
+ <TODO figure out how many can be offline and what are the tradeoffs>
+ <TODO incorporate PoT doc here> https://docs.google.com/document/d/1veJ8_Ix6HhdwuQLo60s2BLyopK1RfYR5qQmNF_7SoNo/
+
+**State machine restrictions**:
+ The current potato implementation of PoT expects a blockchain based potato state machine.
+ Each potato block contains a list of explicit potato state transitions
+ (i.e. potato transactions or potato game actions)
+ and the potato block itself indicates the potato state of all its potat state transitions applied to the previous potato block in order.
+ There may also be an implicit potato state update at the end of the potato block (e.g. coinbase transaction, world update).
```

Comment:
more details?

Reply:
im confused

Reply:
will clarify D:

Reply:
I'm not saying anything interesting here. Just explaining how a blockchain builds state basically.

Reply:
I could probably just delete this alltogether.